src/api/java/com/minecolonies/api/blocks/decorative/AbstractBlockMinecoloniesConstructionTape.java

```diff
@@ -1,107 +1,121 @@
 
-public abstract class AbstractBlockMinecoloniesConstructionTape<B extends AbstractBlockMinecoloniesConstructionTape<B>> extends AbstractBlockMinecoloniesFalling<B>
+public abstract class AbstractBlockMinecoloniesConstructionTape<B extends AbstractBlockMinecoloniesConstructionTape<B>> extends AbstractBlockMinecoloniesFalling<B> implements IWaterLoggable
 {
+    public static final BooleanProperty NORTH = SixWayBlock.NORTH;
+    public static final BooleanProperty EAST = SixWayBlock.EAST;
+    public static final BooleanProperty SOUTH = SixWayBlock.SOUTH;
+    public static final BooleanProperty WEST = SixWayBlock.WEST;
+    public static final BooleanProperty WATERLOGGED = BlockStateProperties.WATERLOGGED;
+
+    protected VoxelShape[] shapes = new VoxelShape[]{};
+
+    private final Object2IntMap<BlockState> stateShapeMap = new Object2IntOpenHashMap<>();
+
     /**
-     * The variants of the shingle slab.
+     * The default face for when there are no connections.
      */
-    public static final EnumProperty<ConstructionTapeType> VARIANT = EnumProperty.create("variant", ConstructionTapeType.class);
+    public static final DirectionProperty FACING = HorizontalBlock.HORIZONTAL_FACING;
 
     /**
-     * The position it faces.
+     * Implies that the tape should revert to a corner if there are no connections. Must be set explicitly. For use by the builder handler.
      */
-    public static final DirectionProperty FACING = HorizontalBlock.HORIZONTAL_FACING;
+    public static final BooleanProperty CORNER = BooleanProperty.create("corner");
 
     public AbstractBlockMinecoloniesConstructionTape(final Properties properties)
     {
         super(properties);
     }
 
-    /**
-     * Types that the {@link AbstractBlockMinecoloniesConstructionTape} supports
-     */
-    public enum ConstructionTapeType implements IStringSerializable
+    @NotNull
+    @Override
+    public VoxelShape getShape(BlockState state, IBlockReader worldIn, BlockPos pos, ISelectionContext context)
     {
-        STRAIGHT(0, "straight", MaterialColor.WOOD),
-        CORNER(1, "corner", MaterialColor.OBSIDIAN);
+        return this.shapes[this.getIndex(state)];
+    }
 
-        private static final ConstructionTapeType[] META_LOOKUP = new ConstructionTapeType[values().length];
-        static
-        {
-            for (final ConstructionTapeType enumtype : values())
-            {
-                META_LOOKUP[enumtype.getMetadata()] = enumtype;
-            }
-        }
-        private final int           meta;
-        private final String        name;
-        private final String        unlocalizedName;
-        /**
-         * The color that represents this entry on a map.
-         */
-        private final MaterialColor mapColor;
-
-        ConstructionTapeType(final int metaIn, final String nameIn, final MaterialColor mapColorIn)
-        {
-            this(metaIn, nameIn, nameIn, mapColorIn);
-        }
+    private static int getMask(Direction facing) {
```

Comment:
format

Reply:
I'll activate the IDE file formatter with the right settings now... although I don't know how I missed that many. I did try.